rt/frontend/jaxrs/src/main/java/org/apache/cxf/jaxrs/utils/ResourceUtils.java

```diff
@@ -939,21 +939,29 @@ public static boolean isValidResourceClass(Class<?> c) {
         }
         return true;
     }
-    
+
+    public static ApplicationPath locateApplicationPath(Class<?> appClass) {
+        ApplicationPath appPath = appClass.getAnnotation(ApplicationPath.class);
+        if (appPath == null && appClass.getSuperclass() != Application.class) {
```

Comment:
I may be wrong here, but solving issue proxies we are going against JAX-RS 2.0 spec:
```
3.6 Annotation Inheritance
... Note that inheritance of class or interface annotations is not supported. ...
```

Does it make sense to limit the scope by identifying the particular use case that CXF dials with proxy (may be using Proxy.isProxyClass or alike) and only then fall back to super class hierarchy?